### PR TITLE
new statusindicator

### PR DESCRIPTION
**Pull request by natibekele**
create new status indicator control to show progress and completion status

### CODE REVIEW

File: src/main/kotlin/org/wycliffeassociates/otter/jvm/app/widgets/statusindicator/StatusIndicator.kt [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+import javafx.scene.control.*
+import javafx.scene.layout.*
+import javafx.scene.paint.*
+import tornadofx.*
+
+class StatusIndicator : Control() {
+
+    var primaryFillProperty: StyleableObjectProperty<Color> =
+            object : SimpleStyleableObjectProperty<Color>(PRIMARY_FILL,
+                    StatusIndicator, "primaryFill", Color.RED) {
```
Comment (jsarabia) [inline src/main/kotlin/org/wycliffeassociates/otter/jvm/app/widgets/statusindicator/StatusIndicator.kt]: can you put each parameter on their own line?

File: src/main/kotlin/org/wycliffeassociates/otter/jvm/app/widgets/statusindicator/StatusIndicator.kt [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+            }
+
+    var primaryFill: Color by primaryFillProperty
+
+
+    var accentFillProperty: StyleableObjectProperty<Color> =
+            object : SimpleStyleableObjectProperty<Color>(ACCENT_FILL,
+                    StatusIndicator, "accentFill", Color.RED) {
```
Comment (jsarabia) [inline src/main/kotlin/org/wycliffeassociates/otter/jvm/app/widgets/statusindicator/StatusIndicator.kt]: see above

File: src/main/kotlin/org/wycliffeassociates/otter/jvm/app/widgets/statusindicator/StatusIndicator.kt [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+            }
+    var accentFill: Color by accentFillProperty
+
+    var trackFillProperty: StyleableObjectProperty<Color> =
+            object : SimpleStyleableObjectProperty<Color>(TRACK_FILL,
+                    StatusIndicator, "trackFill", Color.WHITE) {
```
Comment (jsarabia) [inline src/main/kotlin/org/wycliffeassociates/otter/jvm/app/widgets/statusindicator/StatusIndicator.kt]: see above

File: src/main/kotlin/org/wycliffeassociates/otter/jvm/app/widgets/statusindicator/StatusIndicator.kt [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+                }
+
+                override fun getStyleableProperty(styleable: StatusIndicator): StyleableProperty<Color> {
+                    return styleable.accentFillProperty
+                }
+            }
+
+        private val TRACK_FILL: CssMetaData<StatusIndicator, Color> =
+            object : CssMetaData<StatusIndicator, Color>(
+                "-fx-track-fill",
```
Comment (aunger) [inline src/main/kotlin/org/wycliffeassociates/otter/jvm/app/widgets/statusindicator/StatusIndicator.kt:128]: Should the "accent-fill" lines below match this?

File: src/main/kotlin/org/wycliffeassociates/otter/jvm/app/widgets/statusindicator/StatusIndicatorSkin.kt [hunk 5]
```diff
@@ -0,0 +1,230 @@
...
+        track = StackPane()
+        track.styleClass.setAll("track")
+
+        children.setAll(track, bar)
+
+        control.widthProperty().onChange { observable -> invalidBar = true }
+        control.heightProperty().onChange { observable -> invalidBar = true }
+        control.primaryFillProperty.onChange { observable -> updateBarFill(control.primaryFill, control.accentFill) }
+        control.accentFillProperty.onChange { observable -> updateBarFill(control.primaryFill, control.accentFill) }
+        control.progressProperty.onChange { progress ->
```
Comment (aunger) [inline src/main/kotlin/org/wycliffeassociates/otter/jvm/app/widgets/statusindicator/StatusIndicatorSkin.kt]: I don't have this open in an editor, so I can't test, but should the `observable ->` and `progress ->` stuff above be omitted?

File: src/main/kotlin/org/wycliffeassociates/otter/jvm/app/widgets/statusindicator/StatusIndicatorSkin.kt [hunk 6]
```diff
@@ -0,0 +1,230 @@
...
+        }
+
+        bar = StackPane()
+        track = StackPane()
+        if (localProgressProperty.value <= 1.0000001) {
+            barWidth =
+                ((localProgressProperty.value * width) - snappedLeftInset() - snappedRightInset()).toInt().toDouble() * 2.0 * Math.min(
+                    1.0,
+                    Math.max(0.0, 1.0)
+                ) / 2.0f
```
Comment (aunger) [inline src/main/kotlin/org/wycliffeassociates/otter/jvm/app/widgets/statusindicator/StatusIndicatorSkin.kt]: `2.0 * Math.min(1.0, Math.max(0.0, 1.0)) / 2.0f`
Can be simplified.

Comment (aunger) [reply src/main/kotlin/org/wycliffeassociates/otter/jvm/app/widgets/statusindicator/StatusIndicatorSkin.kt]: Line 224 below too.

Comment (jsarabia) [reply src/main/kotlin/org/wycliffeassociates/otter/jvm/app/widgets/statusindicator/StatusIndicatorSkin.kt]: Math.max(0.0, 1.0) = 1.0
Math.min(1.0, 1.0) = 1.0
2.0 * 1.0 = 2.0
2.0/2.0 = 1.0

What was the point of this at all? There's no variables here?

Comment (natibekele) [reply src/main/kotlin/org/wycliffeassociates/otter/jvm/app/widgets/statusindicator/StatusIndicatorSkin.kt]: It was something that I pulled from the javafx progressbar for calculation the barWidth. I didn't really give it to much attention after I put it in there

**Review by jsarabia — APPROVED** [review]

**Pull request merged**